amqVivaceShortcuts.user.js

```diff
@@ -296,12 +301,24 @@ const optimizedShortcuts = (targets) => {
   let minPos = Infinity;
   let bestSubstring = "";
   let shortcuts = [];
+  let altShortcuts = [];
   let currentLength = 0;
 
-  for (const substring of sortedSubstrings) {
+  for (let substring of sortedSubstrings) {
     const newLength = substring.length;
 
     // Search for longer substrings only if there are not enough shortcuts yet, but display *all* the shortest ones
```

Comment:
```suggestion

    // Search for longer substrings only if there are not enough shortcuts yet, but display *all* the shortest ones
    if (newLength > currentLength && shortcuts.length >= NUM_SHORTCUTS) break;

```

This condition should be checked also before expanding the `shortcuts` array.